packages/@lwc/engine/src/framework/decorators/register.ts

```diff
@@ -48,13 +48,15 @@ export interface RegisterDecoratorMeta {
     readonly publicProps?: PropsDef;
     readonly track?: TrackDef;
     readonly wire?: WireHash;
+    readonly fields?: string[];
 }
 
 export interface DecoratorMeta {
     wire: WireHash | undefined;
     track: TrackDef;
     props: PropsDef;
     methods: MethodDef;
+    fields?: string[];
```

Comment:
I prefer the other annotation, `fields: string[] | undefined`, technically it is almost the same, but the shape of the meta obj should always have the same properties, even if they are set to undefined.